ghga_connector/core/exceptions.py

```diff
@@ -144,8 +151,14 @@ def __init__(self, max_wait_time: int):
 class MaxRetriesReached(RuntimeError, GHGAConnectorException):
     """Thrown, when the specified number of retries has been exceeded."""
 
-    def __init__(self, part_no: int):
-        message = f"Exceeded maximum retries for part number '{part_no}'."
+    def __init__(self, func_name: str, causes: list):
+        # keep track for testing purposes
+        self.num_causes = len(causes)
```

Comment:
Where is this attribute used?

Reply:
Is it only for testing?
In that case it might be even more useful, to just store the causes directly.